hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -481,7 +516,7 @@ private void handleFlush(boolean close)
     checkOpen();
     // flush the last chunk data residing on the currentBuffer
     if (totalDataFlushedLength < writtenDataLength) {
-      final ChunkBuffer currentBuffer = bufferPool.getCurrentBuffer();
+      currentBuffer = bufferPool.getCurrentBuffer();
```

Comment:
Should we also update `currentBufferRemaining`?

Reply:
Yeah, I started to use the newly introduced method which is suggested by the first comment... (update current buffer + remaining)